sql/catalyst/src/main/scala/org/apache/spark/sql/errors/SqlScriptingException.scala

```diff
@@ -17,40 +17,77 @@
 
 package org.apache.spark.sql.errors
 
-import org.apache.spark.SparkException
+import org.apache.spark.{SparkException, SparkThrowableHelper}
+import org.apache.spark.sql.catalyst.trees.Origin
+import org.apache.spark.sql.errors.SqlScriptingException.errorMessageWithLineNumber
+
+class SqlScriptingException protected (
+    origin: Origin,
+    errorClass: String,
+    cause: Throwable,
+    messageParameters: Map[String, String] = Map.empty)
+  extends SparkException(
+    message = errorMessageWithLineNumber(origin, errorClass, messageParameters),
+    errorClass = Option(errorClass),
+    cause = cause,
+    messageParameters = messageParameters
+  ) {
+
```

Comment:
you can omit {}, it's not needed. also, throughout the code we don't leave `)` in the separate row, so it's better to do `messageParameters = messageParameters)` here